File: 1630/arithmetic-subarrays.cpp
class Solution {
public:
    vector<bool> checkArithmeticSubarrays(vector<int>& nums, vector<int>& l, vector<int>& r) {
        //題目的意思, 是 query: l[i] ... r[i] 對應的 nums[l] ... nums[r] 是等差數量
        //所以就暴力試看看吧
        vector<bool> ans;
        for(int i=0; i<l.size(); i++) {
            int a = l[i], b = r[i];
            vector<int> list;
            for(int k=a; k<=b; k++){
                list.push_back(nums[k]);
            }
            sort(list.begin(), list.end());
            int bad = 0;
            for(int k=1; k<list.size(); k++){
                if(list[k]-list[k-1] != list[1]-list[0]) bad = 1;
            }
            if(bad==0) ans.push_back(true);
            else ans.push_back(false);
        }
        return ans;
    }
};

File: 1630/arithmetic-subarrays.py
# 要判斷是否為「等差級數」也就是 nums[i]-nums[i-1] == nums[1]-nums[0]
# 不過題目有稍做修改, index 限定只檢查 l[i]...r[i]範圍
# 因 500個數而已, 暴力應該就算完了
# 但可惡的, 好像需要重新排序的樣子
class Solution:
    def checkArithmeticSubarrays(self, nums: List[int], l: List[int], r: List[int]) -> List[bool]:
        ans = []
        Q = len(r) # 確定問幾次
        for i in range(Q): # 逐筆詢問
            # 題目說 0<=l[i]<r[i]<N 所以不用檢查範圍
            temp = nums[l[i]:r[i]+1] # 取出範圍內的數字
            temp.sort() # 從小到大排好
            diff = temp[1] - temp[0] # 基礎差距
            bad = 0
            for k in range(r[i]-l[i]): # 在小範圍內比較
                if temp[k+1]-temp[k] != diff: # 距離不合
                    bad = 1 # 就壞掉
            if bad == 0: ans.append(True)
            else: ans.append(False)
        return ans
